src/crt/lneg_labs_ldivs_lrems.src

```diff
@@ -0,0 +1,109 @@
+	assume	adl=1
+
+	section	.text
+	public	_labs
+_labs:
+	pop	bc
+	pop	hl
+	pop	de
+	push	de
+	push	hl
+	push	bc
+	bit	7, e
+	ret	z
+
+if PREFER_OS_CRT
+	jp	__lneg
+else
+	require	__lneg
+end if
+
+	section	.text
+	public	__lneg
+
+if PREFER_OS_CRT
+
+__lneg := 0001D0h
+
+else
+
+__lneg:
+	push	af
+	push	de
+	ex	de, hl	; ude=UHL, l=E
+	xor	a, a	; a=0
+	sbc	hl, hl	; uhl=0
+	sbc	hl, de	; uhl=-UHL
+	pop	de	; e=E
+	sbc	a, e
+	ld	e, a	; euhl=-EUHL
+	pop	af
+	ret
+
+end if
+
+	section	.text
+	public	__ldivs
+
+if PREFER_OS_CRT
+
+__ldivs := 0001B4h
+
+else
+	
+__ldivs:
+; I: EUHL=dividend, AUBC=divisor
+; O: euhl=EUHL/AUBC
+
+	rlc	e
+	rrc	e
+	rra
+	adc	a, a
+	push	af
+	push	bc
+
+	call	__ldivs_lrems_common
```

Comment:
I believe that this optimization is invalid:
https://github.com/CE-Programming/toolchain/blob/6c31e81cf6fd67bcae3be65e03e0abf619620f0a/src/crt/ldivs_lrems_common.src#L5-L6